FirebaseSessions/Sources/FirebaseSessions.swift

```diff
@@ -61,7 +75,11 @@ protocol SessionsProvider {
 
     self.initiator.beginListening {
       self.identifiers.generateNewSessionID()
-      self.coordinator.runMain()
+      let event = SessionStartEvent(identifiers: self.identifiers)
+      DispatchQueue.global().async {
+        self.coordinator.beginLoggingSessionStart(event: event) { result in
```

Comment:
Should we call this method as `logSessionStartEvent`? Prefix of `begin` is hard to understand what to expect with that method. I'm generally in favor of using methods with a Verb as a prefix.

Reply:
I used `begin` to convey the fact that it's not always going to log the event. This method needs to take into account settings, sampling, etc.

Reply:
I've been having trouble naming this method - I had `main` before which also has issues

Reply:
Maybe `attemptLoggingSessionStartEvent`